packages/nameguard-js/src/graphemes.ts

```diff
@@ -1,44 +1,578 @@
+  "\u1121",
+  "\ud7e8",
+  "\ud7d7",
+  "\u3139",
+  "\ua963",
+  "\u114c",
+  "\ud7ee",
+  "\u315d",
+  "\u3149",
+  "\u11d3",
+  "\u1162",
+  "\u3146",
+  "\u118f",
+  "\ud7b1",
+  "\u11a4",
+  "\ud7f4",
+  "\ua962",
+  "\u1177",
+  "\u11b7",
+  "\ud7c3",
+  "\u1139",
+  "\ua96a",
+  "\ua97b",
+  "\u110b",
+  "\u119c",
+  "\u11f6",
+  "\u11aa",
+  "\u3170",
+  "\u11f8",
+  "\u3142",
+  "\u11bb",
+  "\u117d",
+  "\u11c3",
+  "\ud7fe",
+  "\u11ac",
+  "\u112b",
+  "\u1109",
+  "\u117a",
+  "\u11eb",
+  "\u1102",
+  "\u1191",
+  "\u1115",
+  "\u3172",
+  "\ud7d2",
+  "\u3151",
+  "\u1199",
+  "\ud7bb",
+  "\u11fb",
+  "\u11b5",
+  "\u1104",
+  "\u3161",
+  "\ua960",
+  "\u111c",
+  "\u11b2",
+  "\u1169",
+  "\u3175",
+  "\u11a0",
+  "\ud7f6",
+  "\u1166",
+  "\u1167",
+  "\u1193",
+  "\ud7eb",
+  "\u119a",
+  "\u1156",
+  "\u3152",
+  "\u1127",
+  "\u315f",
+  "\u115e",
+  "\ua97f",
+  "\u11a8",
+  "\u1194",
+  "\u112c",
+  "\u1180",
+  "\u1107",
+  "\u1152",
+  "\ua97a",
+  "\u1125",
+  "\u111e",
+  "\u112e",
+  "\u3157",
+  "\u1136",
+  "\u11d5",
+  "\u11ae",
+  "\ud7f8",
+  "\u112d",
+  "\u1189",
+  "\ua96f",
+  "\u118e",
+  "\u1154",
+  "\ud7f5",
+  "\u315e",
+  "\u11b4",
+  "\u318a",
+  "\u1183",
+  "\u1175",
+  "\u11ed",
+  "\u11ef",
+  "\u1105",
+  "\u1182",
+  "\ud7ea",
+  "\u1192",
+  "\u11d0",
+  "\ua978",
+  "\u1128",
+  "\u115f",
+  "\u3140",
+  "\u1133",
+  "\u111f",
+  "\u11bf",
+  "\u3134",
+  "\ud7ed",
+  "\u117f",
+  "\ud7ba",
+  "\u110a"
+]);
+
+const INVISIBLE_CHARACTER_JOINERS = new Set([
+  "\u034f",
+  "\u061c",
+  "\u17b4",
+  "\u17b5",
+  "\u200c",
+  "\u200d",
+  "\u200e",
+  "\u200f",
+  "\u206f",
+  "\ufe00",
+  "\ufe01",
+  "\ufe02",
+  "\ufe03",
+  "\ufe04",
+  "\ufe05",
+  "\ufe06",
+  "\ufe07",
+  "\ufe08",
+  "\ufe09",
+  "\ufe0a",
+  "\ufe0b",
+  "\ufe0c",
+  "\ufe0d",
+  "\ufe0e",
+  "\ufe0f",
+  "\ufeff",
+  "\ud834\udd59",
+  "\ud834\udd73",
+  "\ud834\udd74",
+  "\ud834\udd75",
+  "\ud834\udd76",
+  "\ud834\udd77",
+  "\ud834\udd78",
+  "\ud834\udd79",
+  "\ud834\udd7a",
+]);
 
 /**
  * Splits the input string into what users perceive as "characters", called graphemes.
  *
- * An assumption is made that the input string is ENSIP-15 normalized.
+ * This function does not try to be fully compliant with the Unicode grapheme splitting algorithm.
+ * It matches the algorithm used by NameGuard which introduces user-friendly features like Hangul and invisible character splitting.
  *
  * Splitting is performed using the [text-segmentation](https://github.com/niklasvh/text-segmentation) library with added special Hangul treatment.
  * This makes it possible to handle strings with arbitrary Hangul Jamo sequences that most operating systems render as distinct graphemes.
  * Without this fix, some Hangul Jamo would be merged into one grapheme which would seem confusing to the user who sees them as separate.
  * See splitGraphemes.test.ts for examples.
- *
+ * Invisible characters are also split into separate graphemes.
```

Comment:
This is great! Nice job! 🚀 

One other possible idea to document here. What do you think about also documenting the idea that our implementation is safe to use in basically all web browsers and contrasting that with the related browser API for splitting graphemes according to the unicode standard that isn't supported by all browsers today.

This feels like a nice idea for us to also communicate.

Reply:
```
 * This implementation is safe to use in all modern web browsers,
 * unlike the related browser API for splitting graphemes according to the Unicode standard,
 * which isn't supported by all browsers today.
```